nav2_graceful_motion_controller/src/graceful_motion_controller.cpp

```diff
@@ -317,14 +315,19 @@ bool GracefulMotionController::inCollision(const double & x, const double & y, c
 
   // Calculate the cost of the footprint at the robot's current position depending
   // on the shape of the footprint
+  std::unique_ptr<nav2_costmap_2d::FootprintCollisionChecker<nav2_costmap_2d::Costmap2D *>>
+  collision_checker =
```

Comment:
Why was this moved? I think this should be moved back. This isn't a "Free" thing to construct every time.

Reply:
This could be passed to the simulate function without the need for storing as a class member murking up the logic flow, since its only used twice

_Originally posted by @SteveMacenski in https://github.com/ros-planning/navigation2/pull/4021#discussion_r1463968006_

My bad. I think you mentioned here the collision checker instead the costmap_transform_